src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityControllerTrait.php

```diff
@@ -140,18 +141,23 @@ private function handleImport(Request $request, SaveSamlEntityCommand $command)
     private function publishEntity(
         ?ManageEntity $entity,
         SaveEntityCommandInterface $saveCommand,
+        bool $isEntityChangeRequest,
         FlashBagInterface $flashBag
     ) {
         // Merge the save command data into the ManageEntity
         $entity = $this->entityMergeService->mergeEntityCommand($saveCommand, $entity);
 
-        switch ($entity->getEnvironment()) {
-            case Constants::ENVIRONMENT_TEST:
+        switch (true) {
+            case $entity->getEnvironment() === Constants::ENVIRONMENT_TEST:
                 $publishEntityCommand = new PublishEntityTestCommand($entity);
                 $destination = 'entity_published_test';
                 break;
-
-            case Constants::ENVIRONMENT_PRODUCTION:
+            case $isEntityChangeRequest:
```

Comment:
err, this is weird? :thinking: 
Should this not be an if before the switch?

Reply:
_Some background information for what its worth:_

This switch determines what entity edit action should take place. 

Are we editing a test entity, a production entity or maybe edit a published production entity. Previously all of these decisions could be made based on the publication state of the entity. For this new 'change request', which is basically an edit on a published entity, there was need for an additional check. 

To me this is not weird. Before I switched on the entity status. Now that we have two 'hits' on a published entity, I had to add a condition to the switch condition. And my go to trick is to move the conditions to the case statements. And switch on `true`.

Or is this not what you find weird?

If you like, I can explain IRL